app/form/LayerTreeFilter.js

```diff
@@ -0,0 +1,103 @@
+/**
+ * This is the controller for the {@link CpsiMapview.view.form.LayerTreeFilter} component
+ *
+ * @class CpsiMapview.controller.form.LayerTreeFilter
+ */
+Ext.define('CpsiMapview.controller.form.LayerTreeFilter', {
+    extend: 'Ext.app.ViewController',
+
+    alias: 'controller.cmv_layertreefilter',
+
+    /**
+     * Filters the tree by matching the layer's names with the input text.
+     *
+     * @param {Ext.form.field.Text} textField The textfield
+     * @param {String} newVal The new value
+     */
+    onSearch: function (textField, newVal) {
+        var me = this;
+        var view = me.getView();
+
+        // show/hide clear trigger button
+        if (textField.getValue()) {
+            textField.setHideTrigger(false);
+        } else {
+            textField.setHideTrigger(true);
+        }
+
+        var tree = Ext.ComponentQuery.query('cmv_layertree')[0];
+        if (!tree){
+            return;
+        }
+        var store = tree.getStore();
+        if(!store){
+            return;
+        }
+
+        // remove previous filter if exists
+        store.removeFilter(view.TEXT_FILTER_ID);
+
+        var textFilter = new Ext.util.Filter({
+            id: view.TEXT_FILTER_ID,
+            filterFn: function (node) {
+                // always show group layers
+                if (node.hasChildNodes()) {
+                    return true;
+                }
+                // check if baselayers should be filtered
+                if (!view.getDoFilterBaseLayers() && node.getOlLayerProp('isBaseLayer')){
+                    return true;
+                }
+                // enforce case insensitive matches
+                var name = node.getOlLayerProp('name');
```

Comment:
As in comment - perhaps checking against the node text is more user friendly?

Reply:
yes, that works. Done in 7e4acb0

Reply:
Thanks!